app/src/app/services/connectors/authentication/authentication.service.ts

```diff
@@ -0,0 +1,193 @@
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AuthenticationService extends ApiConnector {
+    public currentUser: UserAccount;
+    public get isLoggedIn(): boolean { return this.currentUser && this.currentUser.status == Status.ACTIVE; }
+    private get baseUrl(): string { return environment.integrations.rest_api.url; }
+    public onLogin = new EventEmitter(); // event emitter for admin organization identity pop-up
+
+    constructor(private router: Router, private http: HttpClient, snackbar: MatSnackBar, private restAPIConnector: RestApiConnectorService) { super(snackbar); }
+
+    /**
+     * Check if user is authorized
+     * @param roles list of authorized roles
+     * @returns true, if user is logged in and has an authorized role, false otherwise
+     */
+    public isAuthorized(roles: Role[]): boolean {
+        // is user logged in?
+        if (!this.isLoggedIn) return false;
+        // does user have an authorized role?
+        return roles.indexOf(this.currentUser.role) > -1;
+    }
+
+    /**
+     * Check if user is authorized to edit an object
+     * @param attackType object type
+     * @returns true, if the user can edit the object, false otherwise
+     */
+    public canEdit(attackType?: string): boolean {
+        if (attackType && attackType.includes('collection')) {
+            // restrict collection editing to admin only
+            return this.isAuthorized([Role.ADMIN]);
+        }
+        return this.isAuthorized([Role.EDITOR, Role.ADMIN])
+    }
+
+    /**
+     * User log in sequence
+     * @returns of the logged in user account
+     */
+    public login(): Observable<UserAccount> {
+        return this.getAuthType().pipe(
+            // retrieve authentication configuration
+            concatMap(authnType => {
+                let url = `${this.baseUrl}/authn/${authnType}/login`;
+                if (authnType == "oidc") {
+                    // oidc login
+                    url += `?destination=${encodeURIComponent(window.location.href)}`;
+                    // redirect to OIDC Identity Provider
+                    window.location.href = url;
+                    return this.getSession().pipe(
+                        map(res => { return res; })
+                    );
+                }
+                // anonymous login
+                return this.http.get(url, {responseType: 'text'}).pipe(
+                    concatMap(success => {
+                        return this.getSession().pipe(
+                            map(res => {
+                                this.onLogin.emit();
+                                return res;
+                            })
+                        );
+                    })
+                );
+            }),
+            catchError(this.handleError_raise<UserAccount>()),
+            share()
+        );
+    }
+
+    /**
+     * User logout sequence
+     * Note: this does not log the user out of the OIDC Identity Provider
```

Comment:
Can you edit this note to briefly identify what _does_ happen instead of what _doesn't_ happen?